packages/gensx/src/commands/start.ts

```diff
@@ -0,0 +1,183 @@
+import { existsSync } from "node:fs";
+import { resolve } from "node:path";
+
+import ora from "ora";
+import pc from "picocolors";
+import { watch } from "rollup";
+import { OutputOptions } from "rollup";
+import { Definition } from "typescript-json-schema";
+
+import { createServer } from "../dev-server.js";
+import { getRollupConfig } from "../utils/bundler.js";
+import { getAuth } from "../utils/config.js";
+import { readProjectConfig } from "../utils/project-config.js";
+import { generateSchema } from "../utils/schema.js";
+
+interface StartOptions {
+  project?: string;
+  quiet?: boolean;
+}
+
+interface ServerInstance {
+  stop: () => void;
+}
+
+export async function start(file: string, options: StartOptions) {
+  const quiet = options.quiet ?? false;
+  const spinner = ora({ isSilent: quiet });
+  let currentServer: ServerInstance | null = null;
+  let isFirstBuild = true;
+  // Store schemas outside the event handler
+  let schemas: Record<string, { input: Definition; output: Definition }> = {};
+
+  try {
+    console.info("🔍 Starting GenSX Dev Server...");
+
+    // 1. Validate file exists and is a TypeScript file
+    const absolutePath = resolve(process.cwd(), file);
+    if (!existsSync(absolutePath)) {
+      throw new Error(`File ${file} does not exist`);
+    }
+
+    if (!file.endsWith(".ts") && !file.endsWith(".tsx")) {
+      throw new Error("Only TypeScript files (.ts or .tsx) are supported");
+    }
```

Comment:
Really? Is there a good reason to not support `js` or `jsx`?

Reply:
oh i was under the impression that it had to be typescript so users could tell us the input/output schema for the workflows. Happy to relax this though.